### ISSUE TITLE
PICO-8 startup sound occasionally plays at full volume even when OnionOS volume is set to zero

### ISSUE THREAD

**Issue by schoblaska** [XK9274/pico-8-wrapper-miyoo#8]
I usually keep the volume muted on my device. Maybe about 1/3 of the time when I start the PICO-8 wrapper, the little startup sound plays at full volume. Other than that, volume controls seem to work normally.

I'm on OnionOS v4.2.3 and v0.8.1 of this wrapper.

Thanks for creating this project! Let me know if there's anything I can do to help debug the issue.

**Comment by XK9274** [comment]
Heyo, thanks for dropping in and thankyou!

I know what causes this, will fix in 0.8.2 & update this issue w/ the commit.

Cheers!

**Comment by XK9274** [comment]
Can you give the new launch.sh a go from commit: 5f34aca when you have chance please? It should follow the same logic as volume where it will be checked before opening and set again when mi_ao is restarted, meaning it should always follow on.

I'm unable to test this myself as i've not once seen this happen. The only thing i can think is maybe the volume up key triggers and unmutes? not sure. We should be handling it now in any case.

**Comment by schoblaska** [comment]
I think it worked! I did ten startups with the 0.8.1 version of launch.sh and got four full-volume startup chimes. Then I used the HEAD version you linked, did another ten startups, and got none.

Thanks for the speedy fix!

**Comment by XK9274** [comment]
> I think it worked! I did ten startups with the 0.8.1 version of launch.sh and got four full-volume startup chimes. Then I used the HEAD version you linked, did another ten startups, and got none.
> 
> Thanks for the speedy fix!

Great thanks for testing & getting back to me. When you're confident it's fixed your problem feel free to close the issue, I'll leave it up 👍

**Comment by schoblaska** [comment]
I'll close by 12/8 if I haven't had the issue again by then.

**Comment by XK9274** [comment]
> I'll close by 12/8 if I haven't had the issue again by then.

Sounds good! Cheers & have a good one

**Comment by schoblaska** [comment]
I think I found a new way to reproduce the issue. If I have my volume muted, I don't hear the load startup ping anymore. But if the volume is low (say, three notches above zero), sometimes the startup ping plays at full volume. It seems to happen randomly, about 1/3 of the time like my original issue.

**Comment by XK9274** [comment]
Interesting, is it straight from startup or have you been playing anything else beforehand? (like Drastic for example)

Attached is a launch.sh that will log the volume before (when the current is captured) and after (when it's reset when mi_ao exists again) 

Can you rename it launch.sh (github doesn't allow .sh uploads) and run with it for a few launches, then drop me the file: `/mnt/SDCARD/App/pico/vol_log.txt` when it happens? it would be interesting to see if these values match (they should)

[launch.txt](https://github.com/XK9274/pico-8-wrapper-miyoo/files/13540477/launch.txt)

It will generate something that looks like:

```
Volume set to -17dB, Mute status: 0
Current volume: -17, Current mute: 0
Volume set to -17dB, Mute status: 0
Current volume: -17, Current mute: 0
Volume set to -17dB, Mute status: 0
Current volume: -17, Current mute: 0
Volume set to -17dB, Mute status: 0
```

**Comment by schoblaska** [comment]
I took a [video](https://www.dropbox.com/scl/fi/mlfskpermgooqpytlkb7y/IMG_3743.MOV?rlkey=r3egm5eqhurzubp8nveln3ztu&dl=0) of the issue to hopefully help debug. The first time it starts up with low volume as expected, the second time you can see it boot with the volume much louder. Here's the [volume log](https://github.com/XK9274/pico-8-wrapper-miyoo/files/13547690/vol_log.txt) for these two boots.

```
Current volume: -31, Current mute: 0
Volume set to -31dB, Mute status: 0
Current volume: -31, Current mute: 0
Volume set to -31dB, Mute status: 0
```

**Comment by XK9274** [comment]
RIP, sorry this slipped through the cracks - I've not seen this reported anywhere else so with that in mind, we can modify some things to make sure yours always starts at a set volume, if you'd like (but still honour mute state)

**Comment by schoblaska** [comment]
Nah, if I just have a cursed Miyoo I can deal with it or mess around with the setup myself. I appreciate it, though! Feel free to close this issue :)
